charts/agent/templates/_config-processors.tpl

```diff
@@ -87,3 +87,13 @@ memory_limiter:
 {{- define "config.processors.attributes.observek8sattributes" -}}
 observek8sattributes:
 {{- end -}}
+
+{{- define "config.processors.transform.metrics" -}}
+transform/metrics:
+  metric_statements:
+    - context: metric
+      statements:
+        - set(unit, "test")
```

Comment:
These are my various attempts at making the change show up in o2. All of them have failed so far.
The [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/contexts/ottlmetric) say that I can access the "unit" field from the "metric" context, but apparently it's not the case, unless I am missing something.

Reply:
for some reason even the [metrics transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/metricstransformprocessor/README.md) doesn't mention being able to modify a metric's unit. i wonder if its prohibited for some reason

Reply:
found this example 
https://opentelemetry.io/blog/2023/any-metric-receiver/#fine-tuning-with-the-transform-processor

```
processors:
  transform:
    metric_statements:
      - context: datapoint
        statements:
          - set(metric.unit, attributes["unit"])
          - delete_key(attributes, "unit")
```

Reply:
Ok so this uses a lower level context. This will be more expensive for sure, I'll try it out. Thanks!

Reply:
I will open an issue regarding the `metric` context btw, because it explicitly says that you can access the unit, but it clearly doesn't work.

Reply:
I haven't tried it, just found it in the opentelemetry docs